scripts/js/lib/api/directMapXmlToMdx.test.ts

```diff
@@ -0,0 +1,128 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2025.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import { expect, test } from "@playwright/test";
+import { toMarkdown } from "mdast-util-to-markdown";
+
+import { directMapXmlToMdx } from "./directMapXmlToMdx.js";
+import { xmlParser, collapseWhitespace } from "./xmlToMdx.js";
+
+// Helper function to parse string input and stringify output
+// This is easier than writing and reading the XML and MDX ASTs.
+function directMapXmlToMdxString(xml: string): string {
+  const parsedXml = xmlParser.parse(xml);
+  // console.log(JSON.stringify(parsedXml));
+  const mdastRoot = directMapXmlToMdx(parsedXml);
+  // console.log(JSON.stringify(mdastRoot, null, 2));
+  return collapseWhitespace(toMarkdown(mdastRoot));
+}
+
+function dedentAndTrim(s: string): string {
+  return s.replaceAll("\n        ", "\n").trim();
```

Comment:
Nbd, more an FYI because I was curious how to do it in JS. You can use `" ".repeat(n)`, like how Python has `" " * 10`.